pkg/traceql/ast.go

```diff
@@ -443,6 +443,89 @@ func (s Static) Equals(other Static) bool {
 	return s == other
 }
 
+func (s Static) compare(other *Static) int {
+	if s.Type != other.Type {
+		if s.asFloat() > other.asFloat() {
+			return 1
+		} else if s.asFloat() < other.asFloat() {
+			return -1
+		}
+
+		return 0
+	}
+
+	switch s.Type {
+	case TypeInt:
+		if s.N > other.N {
+			return 1
+		} else if s.N < other.N {
+			return -1
+		}
+	case TypeFloat:
+		if s.F > other.F {
+			return 1
+		} else if s.F < other.F {
+			return -1
+		}
+	case TypeDuration:
+		if s.D > other.D {
+			return 1
+		} else if s.D < other.D {
+			return -1
+		}
+	case TypeString:
+		if s.S > other.S {
+			return 1
+		} else if s.S < other.S {
+			return -1
+		}
+	case TypeBoolean:
+		if s.B && !other.B {
+			return 1
+		} else if !s.B && other.B {
+			return -1
+		}
+	case TypeStatus:
+		if s.Status > other.Status {
+			return 1
+		} else if s.Status < other.Status {
+			return -1
+		}
+	case TypeKind:
+		if s.Kind > other.Kind {
+			return 1
+		} else if s.Kind < other.Kind {
+			return -1
+		}
```

Comment:
Does this comparison make sense for `status` and `kind`? What does a status bigger than another mean?

Reply:
I agree that it's strange and one status isn't really comparable to another status in this way. The only reason I included them was so that if we were to sort Statics like Kind/Status using this function the results would be deterministic.